include/boost/geometry/strategies/cartesian/intersection.hpp

```diff
@@ -68,6 +68,57 @@ namespace boost { namespace geometry
 namespace strategy { namespace intersection
 {
 
+namespace detail_usage
+{
+
+// When calculating the intersection, the information of "a" or "b" can be used.
+// Theoretically this gives equal results, but due to floating point precision
+// there might be tiny differences. These are edge cases.
+// This structure is to determine if "a" or "b" should be used.
+// Prefer the segment closer to the endpoint.
+// If both are about equally close, then prefer the longer segment
+// To avoid hard thresholds, behavior is made fluent.
+// Calculate comparable length indications,
+// the longer the segment (relatively), the lower the value
+// such that the shorter lengths are evaluated higher and will
+// be preferred.
+template <bool IsArithmetic>
+struct use_a
+{
+  template <typename Ct, typename Ev>
+  static bool apply(Ct cla, Ct clb, Ev eva, Ev evb)
```

Comment:
Shouldn't these be `const&` in case these types were user-defined to avoid copies?

Reply:
:heavy_check_mark: